lib/identity_cache/query_api.rb

```diff
@@ -417,7 +417,14 @@ def prefetch_one_association(association, records)
     private
 
     def fetch_recursively_cached_association(ivar_name, association_name) # :nodoc:
-      assoc = if IdentityCache.should_use_cache?
+      association_klass = association(association_name).klass
+      should_use_cache = if association_klass.respond_to?(:should_use_cache?)
```

Comment:
Should we instead include a module that defines this method [as we do with ParentModelExpiration](https://github.com/Shopify/identity_cache/blob/master/lib/identity_cache/configuration_dsl.rb).  That way we wouldn't need a special case for it here and could just use `associated_klass.should_use_cache?` here.